lte/gateway/c/session_manager/ShardTracker.h

```diff
@@ -13,39 +13,39 @@
 #pragma once
 #include <string>
 #include <vector>
+#include <set>
 
 namespace magma {
 
 /* Shards represent groups of UEs placed into buckets of
 a certain size, to make polling more manageable*/
 class ShardTracker {
-  ShardTracker();
-
+ public:
   /**
-   * Add UE to shards based on availability
+   * Add UE to shards based on availability, if the UE already has an
+   * existing shard, return the existing shard id and don't perform
+   * an addition
    * @return index(shard id) where UE was placed
-   * TODO(veshkemburu): Store IMSI as well for easier subscriber reallocation
-   * (GH8167)
    */
-  int add_ue();
+  uint16_t add_ue(std::string imsi);
 
   /**
    * Remove UE from shard
    * @param shard_id location of UE to be removed
    * @return true for successful removal, false for failed removal
    */
-  bool remove_ue(int shard_id);
+  bool remove_ue(std::string imsi, int shard_id);
```

Comment:
shard_id should be unsigned right?

Reply:
Yep and both should be constant. Thanks for the catch.